code/modules/clothing/neck/_neck.dm

```diff
@@ -193,6 +193,32 @@
 	icon_state = "artist"
 	custom_price = 10
 
+/obj/item/clothing/neck/anti_magic_collar
+	name = "anti-magic collar"
+	desc = "A tight collar used on prisoners to restrict their use of magic, while leaving them vulnerable to it's effects"
+	icon_state = "antimagiccollar"
+	resistance_flags = FIRE_PROOF
+
+/obj/item/clothing/neck/anti_magic_collar/equipped(mob/user, slot)
+	. = ..()
+	if(ishuman(user) && slot == ITEM_SLOT_NECK)
+		ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT(type))
```

Comment:
Move this to the proc below since it's the same one

Reply:
updated as requested